parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -140,6 +144,19 @@ retrycmd_get_executable() {
         fi
     done
 }
+retrycmd_curl_file() {
+    curl_retries=$1; wait_sleep=$2; filepath=$3; url=$4
+    echo "${curl_retries} retries"
+    for i in $(seq 1 $curl_retries); do
+        [[ -f $filepath ]] && break
+        if [ $i -eq $curl_retries ]; then
+            return 1
+        else
+            timeout 60 curl -fsSL $url -o $filepath
```

Comment:
should we let timeout be a parameter? I'm not sure if 60s is enough for all cases.

Reply:
can you also add logging for executed times like retrycmd_if_failure

Reply:
TODO: separate PR for cse